pallets/pallet-bonded-coins/src/curves/lmsr.rs

```diff
@@ -34,13 +34,30 @@ where
 	Parameter: FixedSigned + PartialOrd<Precision> + From<Precision> + ToFixed,
 	<Parameter as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign,
 {
-	fn calculate_exp_term(&self, supply: Parameter, x: Parameter) -> Result<Parameter, ArithmeticError> {
-		supply
-			.checked_sub(x)
-			.ok_or(ArithmeticError::Underflow)?
-			.checked_div(self.m)
-			.ok_or(ArithmeticError::DivisionByZero)
-			.and_then(|exponent| exp::<Parameter, Parameter>(exponent).map_err(|_| ArithmeticError::Overflow))
+	fn lse(&self, supply: &[Parameter]) -> Result<Parameter, ArithmeticError> {
+		// Find the maximum value in the supply for numerical stability
+		let max = supply.iter().max().ok_or_else(|| {
+			log::error!(target: LOG_TARGET, "Supply is empty. Found pool with no currencies.");
+			ArithmeticError::Underflow
+		})?;
```

Comment:
if we allow an all-0 supply, we may just treat an empty vector like an all-0 vector instead
```suggestion
		let zero = Parameter::from_num(0);
		let max = supply.iter().max().unwrap_or(&zero);
```

Reply:
https://github.com/KILTprotocol/kilt-node/pull/777#discussion_r1827767284